pages/api/admin/index.ts

```diff
@@ -0,0 +1,52 @@
+import { AchievementCategory } from '@root/constants/achievementInfo';
+import Role from '@root/constants/role';
+import { ValidEnum, ValidObjectId } from '@root/helpers/apiWrapper';
+import withAuth, { NextApiRequestWithAuth } from '@root/lib/withAuth';
+import { AchievementModel } from '@root/models/mongoose';
+import { calcPlayAttempts, refreshIndexCalcs } from '@root/models/schemas/levelSchema';
+import { Types } from 'mongoose';
+import { NextApiResponse } from 'next';
+import { refreshAchievements } from '../stats';
+
+enum AdminCommand {
+  refreshAchievements = 'refreshAchievements',
+  refreshIndexCalcs = 'refreshIndexCalcs',
+  refreshPlayAttempts = 'calcPlayAttempts',
+}
+
+export default withAuth({ POST: {
+  body: {
+    targetId: ValidObjectId(),
+    command: ValidEnum(Object.values(AdminCommand)),
+  }
+} }, async (req: NextApiRequestWithAuth, res: NextApiResponse) => {
+  if (!req.user.roles.includes(Role.ADMIN)) {
+    return res.status(401).json({
+      error: 'Not authorized'
+    });
+  }
+
+  const { targetId, command } = req.body;
+
+  switch (command) {
+  case AdminCommand.refreshAchievements:
+    // clear achievements first to make this idempotent
+    await AchievementModel.deleteMany({ userId: targetId });
```

Comment:
This is a bit scary to me, we could accidentally delete the original time when the achievement was earned. Can we remove this line?

Reply:
Ah... good point about the original time. It was just helpful while testing... We can remove it